### PR TITLE
SNOW-215479 remove unused variables

### PR DESCRIPTION
**Pull request by jameslamb**
Similar to my past few PRs (like #387 ), I've been spending a little time running linters over this project, looking for things I can help contribute.

Today I'm working with `pylint`. This turns up a bunch of things that should be left to maintainers or even ignored: style things like the using snake_case for variable names. However, there are a few things this linter found that I think should be addressed.

<details><summary>pylint command I ran</summary>

```shell
# pylint disables:
#   * C0301: line too long
#   * C0103: snake-case naming
#   * C0114: missing module docstring
#   * C0115: missing class docstring
#   * C0116: missing function docstring
#   * C0302: too many lines in module
#   * C0325: unnecessary parentheses after if / elif
#   * C0330: wrong hanging indent before block
#   * C0413: standard imports before third-part imports
#   * C0415: import outside top level
#   * E0401: unable to import
#   * R0201: method could be a function
#   * R0205: unnecessary object inheritance
#   * R0401: cyclic imports
#   * R0801: similar lines in multiple files
#   * R0902: too many instance attributes
#   * R0903: too few public methods
#   * R0904: too many public methods
#   * R0912: too many branches
#   * R0913: too many arguments
#   * R0914: too many local variables
#   * R0915: too many statements
#   * R1702: too many nested blocks
#   * R1705: unnecessary elif after return
#   * R1710: all return statements should return an expression or none should
#   * R1720: unecessary else after raise
#   * R1723: unnecessary elif
#   * R1724: unnecessary elif after return
#   * R1725: use Python 3 style super()
#   * W0105: string statement has no effect (https://twitter.com/gvanrossum/status/112670605505077248)
#   * W0107: unnecessary pass
#   * W0212: access to protected member
#   * W0511: TODO comments
#   * W0603: using the 'global' statement
#   * W0703: catching too general exception
#   * W1202: use lazy string formatting in logging
pylint \
    --disable=C0103,C0114,C0115,C0116,C0301,C0302,C0325,C0330,C0413,C0415,E0401,R0201,R0205,R0401,R0801,R0902,R0903,R0904,R0912,R0913,R0914,R0915,R1702,R1705,R1710,R1720,R1723,R1724,R1725,W0105,W0107,W0212,W0511,W0603,W0703,W1202 \
    src/snowflake/connector/ \
    > pylint.txt

```

</details>

This PR proposes fixing the first class of issue I found: unused variables.

```text
src/snowflake/connector/network.py:640:22: W0612: Unused variable 'stack_trace' (unused-variable)
src/snowflake/connector/network.py:874:18: W0612: Unused variable 'stack_trace' (unused-variable)
src/snowflake/connector/ocsp_snowflake.py:98:8: W0612: Unused variable 'cls' (unused-variable)
src/snowflake/connector/ocsp_snowflake.py:98:24: W0612: Unused variable 'stack_trace' (unused-variable)
src/snowflake/connector/ocsp_snowflake.py:1155:22: W0612: Unused variable 'cache' (unused-variable)
src/snowflake/connector/incident.py:127:22: W0612: Unused variable 'trace' (unused-variable)
src/snowflake/connector/tool/dump_ocsp_response.py:85:12: W0612: Unused variable 'cert_id' (unused-variable)
```

## How does this PR improve `snowflake-connector-python`?

Removing unnecessary statements makes the functions that call them faster, and reduces the risk of an issue caused by errors raised by that code.

Changing unused variables in return statements to `_` reduces the risk of issues caused by name collisions with other local variables.

## Note to reviewers

I'm planning to submit a few other small PRs with some things found by `pylint`. I'm thinking that breaking them into several small PRs will make them easier to review, but if you disagree and would rather that I bunch things together, let me know and I'd be happy to do that.

Thanks for your time and consideration!

### CODE REVIEW

**Comment by sfc-gh-mkeller** [comment]
Looks good to me, but since this touches OCSP I'd like to get a second opinion from @sfc-gh-hchaturvedi 
I also just kicked off tests!

**Comment by jameslamb** [comment]
I've updated this to get the newest changes on `master`

File: src/snowflake/connector/tool/dump_ocsp_response.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def dump_ocsp_response(urls, output_filename):
         print("Current Time: {}".format(
             strftime('%Y%m%d%H%M%SZ', gmtime(current_time))))
         for issuer, subject in cert_data:
-            cert_id, _ = ocsp.create_ocsp_request(issuer, subject)
+            _, _ = ocsp.create_ocsp_request(issuer, subject)
```
Comment (sfc-gh-hchaturvedi) [inline src/snowflake/connector/tool/dump_ocsp_response.py:86]: interesting thing to see `cert_id` is not being used.

**Review by sfc-gh-hchaturvedi — APPROVED** [review]
Changes LGTM .. Thanks for this ..

**Review by sfc-gh-mkeller — APPROVED** [review]
🚢 🚢

**Comment by sfc-gh-mkeller** [comment]
Closes #402

**Comment by sfc-gh-mkeller** [comment]
Merged through #402

**Pull request closed without merging**